src/components/editor/JSONEditor.tsx

```diff
@@ -25,26 +39,68 @@ export class JSONEditor extends React.Component<IJSONEditorProps, {}> {
 
     private codemirror: ReactCodeMirror.ReactCodeMirror;
 
+    constructor(props: IJSONEditorProps, state: IJSONEditorState) {
+        super(props, state);
+        this.state = {
+            isInError: false,
+        };
+    }
+
     componentDidUpdate(prevProps: IJSONEditorProps) {
         if (prevProps.value !== this.props.value) {
             this.codemirror.getCodeMirror().getDoc().clearHistory();
         }
     }
 
     render() {
+        const classes: string = classNames(
+            'form-group',
+            {
+                'input-validation-error': this.state.isInError,
+            },
+        );
         return (
-            <ReactCodeMirror
-                ref={(codemirror: ReactCodeMirror.ReactCodeMirror) => this.codemirror = codemirror}
-                value={this.props.value}
-                onChange={(json: string) => this.handleChange(json)}
-                options={_.extend({}, JSONEditor.Options, {readOnly: this.props.readOnly})}
-            />
+            <div className={classes}>
+                <ReactCodeMirror
+                    ref={(codemirror: ReactCodeMirror.ReactCodeMirror) => this.codemirror = codemirror}
+                    value={this.props.value}
+                    onChange={(json: string) => this.handleChange(json)}
+                    options={_.extend({}, JSONEditor.Options, {readOnly: this.props.readOnly})}
+                />
+                {this.getValidationDetails()}
+            </div>
         );
     }
 
+    private getValidationDetails(): JSX.Element {
+        return this.state.isInError
+            ? (
+                <div className='input-validation-error-details'>
+                    <span className='input-validation-error-icon'>
+                        <Svg svgName='message-alert' svgClass='icon fill-white' />

```

Comment:
You probably have a svg inside a span inside a span, IIRC you can use className on the Svg instead of wrapping it

Reply:
I refactored `<Svg />` recently and I confirm you can use className directly on the Svg